Sources/MongoSwift/MongoClient.swift

```diff
@@ -66,6 +66,77 @@ public struct MongoClientOptions: CodingStrategyProvider {
     /// Determines whether the client should retry supported write operations (on by default).
     public var retryWrites: Bool?
 
+    /// A type containing options for specifying a MongoDB server API version and related behavior.
+    public struct ServerAPI: Codable {
```

Comment:
so I made this a nested type in line with our recent conversation about nesting types when they're only used in the context of others, but just now I noticed [this section](https://github.com/mongodb/specifications/blob/master/source/versioned-api/versioned-api.rst#overriding-the-declared-api-version) in the spec:

> In the future, we may want to allow users to override the declared API version on a `MongoDatabase`, `MongoCollection`, or individual operation level. However, this is not necessary until there is a different API version and we have data on why and how users would want to override the declared API version.

given that, it seems like maybe it would be a little more future-proof to make `ServerAPI` a standalone type (while still nesting `Version`) and I'm leaning toward doing so.

what do you think?

Reply:
could there be other api settings other than version in the future?

Reply:
Given that warning yeah it probably makes sense to bump it out to the global namespace. We'd need to rename it to `MongoServerAPI` as part of that.

As a side note: my guess is that it will always remain a client option, but you never know how users will want to use things :slightly_smiling_face:

Reply:
> could there be other api settings other than version in the future?
There are already a couple of others I included below as properties on `ServerAPI`, `strict`, and `deprecationErrors`. There could definitely be more settings added to this struct in the future!

I agree it's probably unlikely but better to be safe I think. I have moved it out to a top level type, in its own file, and renamed it `MongoServerAPI`.